pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Dict, Optional, TypeVar, Union
+
+TSaveStorageOptions = TypeVar("TSaveStorageOptions")
+TLoadStorageOptions = TypeVar("TLoadStorageOptions")
+
+
+class CheckpointIOPlugin(ABC):
```

Comment:
"IO" in the name: with the addition of this plugin people will get more ideas how to extend it and will probably put checkpoint logic into it that doesn't have to do with IO. What are your thoughts, is it good to name it like this, or maybe go with just CheckpointPlugin?

Reply:
I like that, to keep this plugin focused!

Reply:
i'm open to names, but @SeanNaren covered the reasons perfectly.

with larger models imminent, i'd anticipate more people wanting to optimize this specific portion of the checkpoint saving/loading procedure, so carving it out as a dedicated plugin affords that space.

this is still gated by its usage in the training type. even if someone wants to write a whole custom strategy and a whole custom IO plugin, these remain bound by when the trainer calls them